filter.go

```diff
@@ -44,3 +44,18 @@ func Distinct[T comparable](source []T) []T {
 
 	return MapKeys(set)
 }
+
+// Difference finds a set difference between a and b
+// (values that are in a but not in b or a-b).
+func Difference[T comparable](a []T, b []T) []T {
+	var (
+		ok   bool
+		bMap = SliceToMap(b, func(t T) T { return t })
+	)
+
+	return FilterBy(a, func(t T) bool {
+		_, ok = bMap[t]
+		// we need a elements that are not in B
+		return !ok
+	})
+}
```

Comment:
Can be rewritten as:
```golang
func Difference[T comparable](a []T, b []T) []T {
  var notInB = func(item T) bool { return !InFilter(b))

  return return FilterBy(a, notInB)
}
```

Reply:
Are you sure? `InFilter` does not even return a boolean, it's a filter function

Reply:
fixed.
```golang
func Difference[T comparable](a []T, b []T) []T {
	var notInB = func(item T) bool { return !InFilter(b)(item) }

	return FilterBy(a, notInB)
}
```

However this approach is not effective - InFilter(b) is called multiple times instea

Reply:
not sure this adds anything tgh, maybe leave as is?